packages/core-mobile/app/store/notifications/listeners/subscribeBalanceChangeNotifications.ts

```diff
@@ -8,11 +8,28 @@ import { subscribeForBalanceChange } from 'services/notifications/balanceChange/
 import Logger from 'utils/Logger'
 import { ChannelId } from 'services/notifications/channels'
 import NotificationsService from 'services/notifications/NotificationsService'
+import { AnyAction, PayloadAction } from '@reduxjs/toolkit'
+import { setFeatureFlags } from 'store/posthog'
+import { FeatureFlags, FeatureGates } from 'services/posthog/types'
 
 export async function subscribeBalanceChangeNotifications(
+  action: AnyAction,
   listenerApi: AppListenerEffectAPI
 ): Promise<void> {
   const { getState } = listenerApi
+
+  if (action.type === setFeatureFlags.type) {
+    const setFeatureFlagsAction = action as PayloadAction<FeatureFlags>
+    if (
+      !setFeatureFlagsAction.payload[FeatureGates.BALANCE_CHANGE_NOTIFICATIONS]
+    ) {
+      Logger.info(
+        'skipping subscribeBalanceChangeNotifications: feature flag disabled'
+      )
+      return
+    }
+  }
```

Comment:
in the isAnyOf you can create a function that returns true only when setFeatureFlagsAction.payload[FeatureGates.BALANCE_CHANGE_NOTIFICATIONS] is on. that way you don't have to add the logic here to skip.

Reply:
[Refactor notification listeners](https://github.com/ava-labs/core-mobile/pull/1659/commits/11e25b005f6c490ba5d2969cead6d8c3c2b0d42d)